### PR TITLE
Finish implementing scoring in GameManager

### PR DESCRIPTION
**Pull request by strburst**
`GameManager` is now capable of fully scoring a game (via `scoreGame`). Consult the Javadocs/code for the scoring formula.

### CODE REVIEW

File: src/main/java/io/github/mountainrange/mule/gameplay/Grid.java [hunk 1]
```diff
@@ -207,4 +209,35 @@ public void removePlayer() {
 	public Point2D getPlayerPosition() {
 		return playerPosition;
 	}
+
+	@Override
```
Comment (jgkamat) [inline src/main/java/io/github/mountainrange/mule/gameplay/Grid.java:213]: Just a heads up, this iterator will only go over the tiles, not the players. We decided to store the player independently for the tiles so they could eventually move around freely. Good idea though!

**Comment by jgkamat** [comment]
LGTM :+1:

File: src/main/java/io/github/mountainrange/mule/gameplay/Grid.java (hunk 1 above)
Comment (strburst) [inline src/main/java/io/github/mountainrange/mule/gameplay/Grid.java:213]: Yes, it should only iterate over tiles. I needed a convenient way to give players points for each tile they owned (hence implementing `Iterable<Tile>`).

Comment (jgkamat) [inline src/main/java/io/github/mountainrange/mule/gameplay/Grid.java:213]: Ah, makes sense, cool!

**Pull request merged**